rudder-core/src/main/scala/com/normation/rudder/services/quicksearch/QuickSearchBackendImpl.scala

```diff
@@ -404,25 +410,31 @@ object QSLdapBackend {
        * Returns Option[(attribute name, matching value)]
        */
       def matchValue(a: Attribute, token: String, pattern: Pattern): Option[(String, String)] = {
+
         //test a matcher against values
-        def findValue(test: String => Boolean): Option[(String, String)] = {
-          a.getValues.find(test).map(v => (a.getName, v))
+        def findValue(values: Array[String], test: String => Boolean): Option[(String, String)] = {
+          values.find(test).map(v => (a.getName, v))
         }
         //test a list of pattern matcher against values
         def testAndFindValue(testers: List[String => Boolean]): Option[(String, String)] = {
-          testers.find(f => f(token)).flatMap(findValue)
+          testers.find(f => f(token)).flatMap(findValue(a.getValues, _))
         }
 
         if(a.getName == A_OC) {
-          None
+          //here we must just avoid "top" and "kind" object class
+          //and also remove the "node" part for "linuxNode", etc
+          val values = a.getValues.flatMap(s =>
+            if(ocRudderTypes.contains(s)) None else Some(s.replace("Node", ""))
```

Comment:
i'm not really fond of the twice hardcoded "Node" - could you at least use a constant ?